library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java

```diff
@@ -690,15 +697,15 @@ else if (response == Constants.BILLING_RESPONSE_RESULT_ITEM_ALREADY_OWNED)
 						loadOwnedPurchasesFromGoogle();
 					}
 					TransactionDetails details = getPurchaseTransactionDetails(productId);
-					if (!checkMerchant(details))
+					if (!checkMerchant(details)) // TODO will throw NPE if details is null
```

Comment:
Rather than try to "fix" this here, I'm leaving existing behavior in place and will investigate in the future.